src/crawlee/crawlers/_basic/_basic_crawler.py

```diff
@@ -1265,3 +1307,30 @@ def _check_request_collision(self, request: Request, session: Session | None) ->
             raise RequestCollisionError(
                 f'The Session (id: {request.session_id}) bound to the Request is no longer available in SessionPool'
             )
+
+    async def _is_allowed_based_on_robots_txt_file(self, url: str) -> bool:
+        """Check if the URL is allowed based on the robots.txt file.
+
+        Args:
+            url: The URL to check.
+        """
+        if not self._respect_robots_txt_file:
+            return True
+        robots_txt_file = await self._get_robots_txt_file_for_url(url)
+        return not robots_txt_file or robots_txt_file.is_allowed(url)
+
+    async def _get_robots_txt_file_for_url(self, url: str) -> RobotsTxtFile | None:
```

Comment:
I believe we should use some synchronization mechanism so that we don't fetch the same robots.txt file multiple times in parallel.